FreeRTOS/Demo/WIN32-MingW/main.c

```diff
@@ -134,204 +159,221 @@ declared here, as a global, so it can be checked by a test that is implemented
+    /* This function will be called once only, when the daemon task starts to
+    execute (sometimes called the timer task). This is useful if the
+    application includes initialisation code that would benefit from executing
+    after the scheduler has been started. */
 }
 /*-----------------------------------------------------------*/
 
 void vAssertCalled( unsigned long ulLine, const char * const pcFileName )
 {
-static BaseType_t xPrinted = pdFALSE;
 volatile uint32_t ulSetToNonZeroInDebuggerToContinue = 0;
 
-	/* Called if an assertion passed to configASSERT() fails.  See
-	http://www.freertos.org/a00110.html#configASSERT for more information. */
-
-	/* Parameters are not used. */
-	( void ) ulLine;
-	( void ) pcFileName;
-
-
- 	taskENTER_CRITICAL();
-	{
-		/* Stop the trace recording. */
-		if( xPrinted == pdFALSE )
-		{
-			xPrinted = pdTRUE;
-			if( xTraceRunning == pdTRUE )
-			{
-				prvSaveTraceFile();
-			}
-		}
-
-		/* You can step out of this function to debug the assertion by using
-		the debugger to set ulSetToNonZeroInDebuggerToContinue to a non-zero
-		value. */
-		while( ulSetToNonZeroInDebuggerToContinue == 0 )
-		{
-			__asm volatile( "NOP" );
-			__asm volatile( "NOP" );
-		}
-	}
-	taskEXIT_CRITICAL();
+    /* Called if an assertion passed to configASSERT() fails.  See
+    http://www.freertos.org/a00110.html#configASSERT for more information. */
+
+    /* Parameters are not used. */
+    ( void ) ulLine;
+    ( void ) pcFileName;
+
+
+    taskENTER_CRITICAL();
+    {
+        printf("ASSERT! Line %ld, file %s, GetLastError() %ld\r\n", ulLine, pcFileName, GetLastError());
+        fflush( stdout );
+
+        /* Stop the trace recording. */
+        ( void ) xTraceDisable();
+        prvSaveTraceFile();
+
+        /* You can step out of this function to debug the assertion by using
+        the debugger to set ulSetToNonZeroInDebuggerToContinue to a non-zero
+        value. */
+        while( ulSetToNonZeroInDebuggerToContinue == 0 )
+        {
+            __asm volatile( "NOP" );
+            __asm volatile( "NOP" );
+        }
+
+        /* Re-enable recording */
+        ( void ) xTraceEnable( TRC_START );
+    }
+    taskEXIT_CRITICAL();
 }
 /*-----------------------------------------------------------*/
 
 static void prvSaveTraceFile( void )
 {
-	/* Tracing is not used when code coverage analysis is being performed. */
-	#if( projCOVERAGE_TEST != 1 )
-	{
-		FILE* pxOutputFile;
-
-		vTraceStop();
-
-		pxOutputFile = fopen( "Trace.dump", "wb");
-
-		if( pxOutputFile != NULL )
-		{
-			fwrite( RecorderDataPtr, sizeof( RecorderDataType ), 1, pxOutputFile );
-			fclose( pxOutputFile );
-			printf( "\r\nTrace output saved to Trace.dump\r\n" );
-		}
-		else
-		{
-			printf( "\r\nFailed to create trace dump file\r\n" );
-		}
-	}
-	#endif
+    /* Tracing is not used when code coverage analysis is being performed. */
+    #if( projCOVERAGE_TEST != 1 )
+    {
+        FILE* pxOutputFile;
+
+        vTraceStop();

```

Comment:
Looking through the ways this function gets called, does this call to TraceStop() need to be here?
It seems like the ways to end up in this function are wrapped with TraceDisable()/TraceEnable() calls already?

Reply:
ya that shouldn't be there. doesn't hurt anything if the trace is already stopped, but I will remove it.